platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/result/DefaultResolvedArtifactResult.java

```diff
@@ -24,13 +25,16 @@
 import org.gradle.internal.DisplayName;
 
 import java.io.File;
+import java.util.Collections;
 import java.util.List;
+import java.util.Set;
 
 public class DefaultResolvedArtifactResult implements ResolvedArtifactResult {
     private final ComponentArtifactIdentifier identifier;
     private final ResolvedVariantResult variant;
     private final Class<? extends Artifact> type;
     private final File file;
+    private final Set<ResolvedVariantResult> variants;
```

Comment:
🤔 The type here could be `ImmutableSet`, then the `getVariants()` method could return that directly, giving more information to the caller and removing the possibility of 2 different types being used depending on construction.